core/src/main/java/org/hibernate/ogm/hibernatecore/impl/BackendCustomLoader.java

```diff
@@ -182,8 +184,23 @@ public BackendCustomLoaderContext(QueryableGridDialect<T> gridDialect, BackendCu
 			);
 		}
 
-		public ClosableIterator<Tuple> executeQuery(QueryParameters queryParameters) {
-			return gridDialect.executeBackendQuery( query, queryParameters );
+		public ClosableIterator<Tuple> executeQuery(SessionImplementor session, QueryParameters queryParameters) {
+			TransactionCoordinator coordinator = session.getTransactionCoordinator();
+			return gridDialect.executeBackendQuery( query, queryParameters, new BackendQueryContext( coordinator ) );
+		}
+	}
+
+	private static class BackendQueryContext implements QueryContext {
+
+		private final TransactionCoordinator coordinator;
+
+		public BackendQueryContext(TransactionCoordinator coordinator) {
+			this.coordinator = coordinator;
```

Comment:
I'm pretty confident :)
I guess it could be null in case a user define a custom transaction coordinator builder.

Anyway, this is not a problem for most of the dialect, but Neo4j Remote needs to check that it is not null and throw a meaningful exception. I will add an assert like you suggested before